app/javascript/react/containers/RestaurantReviewFormContainer.js

```diff
@@ -0,0 +1,161 @@
+    this.submitReview = this.submitReview.bind(this);
+    this.handleFormSubmit = this.handleFormSubmit.bind(this);
+    this.validateReviewTitle = this.validateReviewTitle.bind(this);
+    this.validateReviewBody = this.validateReviewBody.bind(this);
+    this.validateReviewRating = this.validateReviewRating.bind(this);
+  }
+
+  handleFormSubmit(event) {
+    event.preventDefault();
+    let validTitle = this.validateReviewTitle(this.state.reviewTitle)
+    let validBody = this.validateReviewBody(this.state.reviewBody)
+    let validRating = this.validateReviewRating(this.state.reviewRating)
+
+    if (validTitle && validBody && validRating){
+      let formPayLoad = {
+        title: this.state.reviewTitle,
+        body: this.state.reviewBody,
+        rating: this.state.reviewRating,
+        restaurant_id: this.props.params.restaurant_id
+      };
+      this.submitReview(formPayLoad);
+  }
+}
+
+  validateReviewTitle(selection) {
+    if (selection.trim() === '') {
+      console.log("title error")
+      let newError = { reviewTitle: 'You must enter a Title!' }
+      this.setState({ errors: Object.assign(this.state.errors, newError) })
+      return false
+    } else {
+      let errorState = this.state.errors
+      delete errorState.reviewTitle;
+      this.setState({ errors: errorState })
+      return true
+    }
+  }
+
+  validateReviewBody(selection) {
+    if (selection.trim() === '') {
+      console.log("body error")
+      let newError = { reviewBody: 'You must enter a Description!' }
+      this.setState({ errors: Object.assign(this.state.errors, newError) })
+      return false
+    } else {
+      let errorState = this.state.errors
+      delete errorState.reviewBody;
+      this.setState({ errors: errorState })
+      return true
+    }
+  }
+
+  validateReviewRating(selection) {
+    if (selection.trim() === '') {
+      console.log("rating error")
+      let newError = { reviewRating: 'You must enter a Rating!' }
+      this.setState({ errors: Object.assign(this.state.errors, newError) })
+      return false
+    } else {
+      let errorState = this.state.errors
+      delete errorState.reviewRating;
+      this.setState({ errors: errorState })
+      return true
+    }
+  }
+
+
+  submitReview(payload) {
+    let post_url = `/api/v1/reviews`
+    let data = JSON.stringify(payload)
+    fetch(post_url, {
+      method: 'POST',
+      credentials: 'include',
+      headers: {
+        'Accept': 'application/json',
+        'Content-Type': 'application/json',
+      },
+      body: data
+    })
+    .then(response => {
+      if (response.status == 500) {
+        console.log("500 code")
+        return response.json()
+      } else {
+        console.log("Implement Redirect")
+        return response.json()
+      }
+    }).then(parsedBody => {
+      this.setState({ errors: Object.assign(this.state.errors, parsedBody.error_list) })
+      console.log(parsedBody)
+    })
+
+    console.log("submit review")
+  }
```

Comment:
Oof. Please use error handling that is outlined the fetch exercise with `.catch`

Reply:
added proper error handling.